mirgecom/artificial_viscosity.py

```diff
@@ -0,0 +1,325 @@
+r""":mod:`mirgecom.artificial_viscosity` Artificial viscocity for Euler.
+
+Euler Equations with artificial viscosity term:
+
+.. math::
+
+    \partial_t \mathbf{Q} = -\nabla\cdot\mathbf{F}^I +
+    \nabla\cdot{\varepsilon\nabla\mathbf{Q}}
+
+where:
+
+-  fluid state: $\mathbf{Q} = [\rho, \rho{E}, \rho\mathbf{V}, \rho\mathbf{Y}]$
+-  inviscid fluxes: $\mathbf{F}^I$
+-  artifical viscosity coefficient: $\varepsilon$
+
+To evalutate the second order derivative the problem is recast as a set of first
+ order problems:
+
+.. math::
+
+    \partial_t{\mathbf{Q}} &= \nabla\cdot\mathbf{R} -\nabla\cdot\mathbf{F}^I \\
+    \mathbf{R} &= \varepsilon\nabla\mathbf{Q}
+
+where $\mathbf{R}$ is an intermediate variable, and the artitifial viscosity
+coefficient, $\varepsilon$, is spatially dependent and calculated using the
+smoothness indicator of [Persson_2012]_:
+
+.. math::
+
+    S_e = \frac{\langle u_{N_p} - u_{N_{p-1}}, u_{N_p} -
+        u_{N_{p-1}}\rangle_e}{\langle u_{N_p}, u_{N_p} \rangle_e}
+
```

Comment:
Should this stuff be reworded post-generalization?

Reply:
https://github.com/w-hagen/mirgecom/pull/18 @MTCam